File: backend-jakarta/src/main/java/com/example/backendjakarta/SQLService.java
package com.example.backendjakarta;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class SQLService {
    private SQLService() {}

    static boolean isInstantiated = false;
    static Connection connection;

    public static void instantiate() {
        if (!isInstantiated) {
            return;
        }
        try {
            Class.forName("org.postgresql.Driver");
        } catch (ClassNotFoundException e) {
            throw new RuntimeException(e);
        }
        try {
            connection = DriverManager.getConnection(
            "jdbc:postgresql://localhost:5432/mydb",
            "pgadmin",
            "pgadmin"
            );
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    public static Connection getConnection() {
        instantiate();
        return connection;
    }
}

File: backend-jakarta-ee/src/main/kotlin/com/example/backendjakartaee/SqlService.kt
package com.example.backendjakartaee

import java.sql.Connection
import java.sql.DriverManager
import io.github.cdimascio.dotenv.dotenv

object SqlService {
    private var isInstantiated = false
    private lateinit var connection: Connection

    private fun instantiate() {
        if (isInstantiated) {
            return
        }
        Class.forName("org.postgresql.Driver")

        val env = dotenv()

        connection = DriverManager.getConnection(
            env["DB_URL"],
            env["DB_USERNAME"],
            env["DB_PASSWORD"]
        )
    }

    fun getConnection(): Connection {
        instantiate()
        return connection
    }
}

